addon/scheduler.ts

```diff
@@ -80,12 +71,7 @@ export function emitMark() {
 export function beginTransition(): void {
   if (_whenRouteDidChange.isResolved) {
     _whenRouteDidChange = _defer(APP_SCHEDULER_LABEL);
-    _whenRoutePainted = _whenRouteDidChange.promise.then(() =>
-      _afterNextPaint(_whenRoutePaintedScheduleFn)
-    );
-    _whenRouteIdle = _whenRoutePainted.then(() =>
-      _afterNextPaint(_whenRouteIdleScheduleFn)
-    );
+    _whenRouteIdle = _whenRouteDidChange.promise;
```

Comment:
I think we still need a macro task here, either a rAF or a setTimeout (otherwise we have no real guarantee that things have painted in the browser). I personally like a rAF, something like this:

```suggestion
    _whenRouteIdle = _whenRouteDidChange.promise.then(() => {
      let nextPaintToken = waiter.beginAsync();
      
      return new Promise(requestAnimationFrame).then(() => {	
        waiter.endAsync(nextPaintToken);	
      });
    });
```

Reply:
Discussed with @scalvert and realized that `_whenRouteDidChange.promise` is _actually_ only resolved after a afterRender + rAF + rAF. Disregard (sorry about the noise)!